tests/bofire/outlier_detection/test_outlier_detection.py

```diff
@@ -39,26 +43,253 @@ def test_IterativeTrimming():
+
+
+def test_outlier_detectors_check_compatibility():
+    data_model1 = data_models.IterativeTrimming(
+        base_gp=SingleTaskGPSurrogate(
+            inputs=Inputs(
+                features=[
+                    ContinuousInput(
+                        key=f"x_{i+1}",
+                        bounds=(-4, 4),
+                    )
+                    for i in range(3)
+                ]
+                + [
+                    CategoricalDescriptorInput(
+                        key="cat",
+                        categories=["apple", "banana"],
+                        descriptors=["length", "width"],
+                        values=[[1, 2], [3, 4]],
+                    )
+                ]
+            ),
+            outputs=Outputs(features=[ContinuousOutput(key="y")]),
+            scaler=ScalerEnum.NORMALIZE,
+            input_preprocessing_specs={"cat": CategoricalEncodingEnum.ONE_HOT},
+        )
+    )
+    data_model2 = data_models.IterativeTrimming(
+        base_gp=SingleTaskGPSurrogate(
+            inputs=Inputs(
+                features=[
+                    ContinuousInput(
+                        key=f"x_{i+1}",
+                        bounds=(-4, 4),
+                    )
+                    for i in range(2)
+                ]
+                + [
+                    CategoricalDescriptorInput(
+                        key="cat",
+                        categories=["apple", "banana"],
+                        descriptors=["length", "width"],
+                        values=[[1, 2], [3, 4]],
+                    )
+                ]
+            ),
+            outputs=Outputs(features=[ContinuousOutput(key="y2")]),
+            scaler=ScalerEnum.NORMALIZE,
+            input_preprocessing_specs={"cat": CategoricalEncodingEnum.ONE_HOT},
+        )
+    )
+    data_model = data_models.OutlierDetections(detectors=[data_model1, data_model2])
+    # models = OutlierDetections(data_model=data_model)
+    # check too less input features
+    inp = Inputs(
+        features=[
+            ContinuousInput(
+                key=f"x_{i+1}",
+                bounds=(-4, 4),
+            )
+            for i in range(3)
+        ]
+    )
+    out = Outputs(features=[ContinuousOutput(key="y"), ContinuousOutput(key="y2")])
+    with pytest.raises(ValueError):
+        data_model._check_compability(inp, out)
+    # check unused input features
+    inp = Inputs(
+        features=[
+            ContinuousInput(
+                key=f"x_{i+1}",
+                bounds=(-4, 4),
+            )
+            for i in range(4)
+        ]
+        + [
+            CategoricalDescriptorInput(
+                key="cat",
+                categories=["apple", "banana"],
+                descriptors=["length", "width"],
+                values=[[1, 2], [3, 4]],
+            )
+        ]
+    )
+    out = Outputs(features=[ContinuousOutput(key="y"), ContinuousOutput(key="y2")])
+    with pytest.raises(ValueError):
```

Comment:
```suggestion
    with pytest.raises(ValueErro, message=...):
```

Can you also add here the messages of the individual errors raised, I know it is not done everywhere, but I try to force myself to catch it in new parts that are added.

Reply:
Do it then for the whole test.